x-pack/plugins/endpoint/server/services/resolver/phase0_builder.test.ts

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { ResolverData } from '../../../common/types';
+import { EventBuilder } from './event_builder.test';
+
+export class Phase0Builder implements EventBuilder {
+  constructor(
+    public readonly endpointID: string,
+    public readonly originEntityID: number,
+    public readonly originParentEntityID: number
+  ) {}
+  buildEvent(entityID: number, parentEntityID: number): ResolverData {
+    return {
+      endgame: {
+        event_type_full: 'process_event',
+        event_subtype_full: 'creation_event',
+        unique_pid: entityID,
+        unique_ppid: parentEntityID,
+      },
+      agent: {
+        id: this.endpointID,
+      },
+    };
+  }
+  startingChildrenEntityID(): number {
+    // +1 so the children's entity ids start after the origin's
+    return this.originEntityID + 1;
+  }
```

Comment:
I'm not quite sure why this is necessary--children nodes should have a `unique_ppid` value that matches this node's entity id, no need to figure out what a "starting child id" is, especially given that we don't want to require entity_id values to be monotonically increasing.

Reply:
Yeah this is kind of ugly, I'll refactor it. This is just a test class. I use it to dynamically create nodes so I can run tests. There's two reasons that the child node's entity ID needs to be unique, first I wanted to avoid creating two child nodes that have the same entity ID and second I wanted the entity ID to not overlap with the parent's entity ID, otherwise the nodes willed be grouped together in the response and it will throw off the tests. If the origin's entity ID is 5 and we create 10 child nodes, if their entity ID starts at 0, it would go 0 - 9, and then have the origin with entity ID 5 and child node with the same entity ID, which would cause a collision.